### ISSUE TITLE
Visual code request

### ISSUE THREAD

**Issue by cxyLenbai** [Jiaxin-Lu/Jigsaw#12]
Hello, I would like to visually check the situation of the predicted fracture edge line of the fragments. Can you provide relevant codes?

**Comment by Nes-tle** [comment]
You may refer to "cls_logits" or "cls_pred" of the out_dict (See model/jigsaw
/joint_seg_align_model.py). These fields indicate whether each point its predicted as a fracture point. You may customize your visualization with any preferred renderer.

**Comment by cxyLenbai** [comment]
Hello, I would like to know that I checked the contents of cls_pred according to the method you said, but I found that I sampled 5000 points in the whole model, but why are there 10000 contents composed of 0 and 1 in cls_pred? Shouldn't there be 5,000 zeros and ones? I attached the point cloud I collected and the output cls_pred file.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Jiaxin-Lu/Jigsaw"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月15日(星期五) 下午3:00
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Jiaxin-Lu/Jigsaw] Visual code request (Issue #12)





 
You may refer to "cls_logits" or "cls_pred" of the out_dict (See model/jigsaw
 /joint_seg_align_model.py). These fields indicate whether each point its predicted as a fracture point. You may customize your visualization with any preferred renderer.

**Comment by Hazel-up** [comment]
> Hello, I would like to visually check the situation of the predicted fracture edge line of the fragments. Can you provide relevant codes?

I saw your question about visualizing the predicted fracture edge lines. I'm currently facing a similar challenge and was wondering if you managed to successfully implement the visualization.

If you did, could you please share how you approached it or provide any tips? It would be really helpful!

**Comment by cxyLenbai** [comment]
Hello, I have actually visualized the edge result now, but the effect is not good. I visualized the real label label calculated by the algorithm on the point cloud, and the effect is not satisfactory. I don't know if it is because the threshold is not set properly or because the idea of calculating the real label in the paper is not correct. I actually have doubts about the method of calculating the real label in the paper, do you think?




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月22日(星期天) 下午5:15
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Jiaxin-Lu/Jigsaw] Visual code request (Issue #12)















您好，我想直观地检查碎片预测断裂边缘线的情况。可以提供相关的代码吗？

我看到了你关于可视化预测的断裂边缘线的问题。我目前正面临类似的挑战，想知道你们是否成功实现了可视化。

如果是这样，请分享一下你是如何处理的，或者提供一些建议。这会很有帮助的！

-
直接回复此邮件，在GitHub上查看它，或取消订阅.
***@***.*** &gt;

**Comment by cxyLenbai** [comment]
At present, I have realized the visualization of the edge in the paper method, and confirmed that the method in the paper is correct.




------------------ 原始邮件 ------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月22日(星期天) 下午5:15
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Jiaxin-Lu/Jigsaw] Visual code request (Issue #12)















您好，我想直观地检查碎片预测断裂边缘线的情况。可以提供相关的代码吗？

我看到了你关于可视化预测的断裂边缘线的问题。我目前正面临类似的挑战，想知道你们是否成功实现了可视化。

如果是这样，请分享一下你是如何处理的，或者提供一些建议。这会很有帮助的！

-
直接回复此邮件，在GitHub上查看它，或取消订阅.
***@***.*** &gt;

**Comment by 12019100748** [comment]
@cxyLenbai How did you achieve the visualization of the edges in the paper methodology, and can you share the code? Thank you！
